src/OpenToolkit.Mathematics/Geometry/Box2.cs

```diff
@@ -121,6 +121,160 @@ public Vector2 Center
             set => Translate(value - Center);
         }
 
+        // --
+
+        public float Width
+        {
+            get => _max.X - _min.X;
+            set => _max.X = _min.X + value;
+        }
+
+        public float Height
+        {
+            get => _max.Y - _min.Y;
+            set => _max.Y = _min.Y + value;
+        }
+
+        public float Left
+        {
+            get => _min.X;
+            set => _min.X = value;
+        }
+
+        public float Top
+        {
+            get => _min.Y;
+            set => _min.Y = value;
+        }
+
+        public float Right
+        {
+            get => _max.X;
+            set => _max.X = value;
+        }
+
+        public float Bottom
+        {
+            get => _max.Y;
+            set => _max.Y = value;
+        }
+
+        public float X
+        {
+            get => _min.X;
+            set => _min.X = value;
+        }
+
+        public float Y
+        {
+            get => _min.Y;
+            set => _min.Y = value;
+        }
+
+        public float SizeX
+        {
+            get => _max.X - _min.X;
+        }
+
+        public float SizeY
+        {
+            get => _max.Y - _min.Y;
+        }
+
+        public Vector2 Location => _min;
+
+        public bool IsEmpty => _min.X == 0 && _min.Y == 0 && _max.X == 0 && _max.Y == 0;
+
+        public static readonly Box2 Zero = new Box2(0, 0, 0, 0);
+
+        public static readonly Box2 Identity = new Box2(0, 0, 1, 1);
+
+        public static Box2 FromSize(Vector2 location, Vector2 size)
+        {
+            return new Box2(location, location + size);
+        }
+
+        public static Box2 FromPositions(Vector2 min, Vector2 max)
+        {
+            return new Box2(min, max);
+        }
+
+        public static Box2 FromPositions(float minX, float minY, float maxX, float maxY)
+        {
+            return new Box2(minX, minY, maxX, maxY);
+        }
+
+        public void Intersect(Box2 other)
```

Comment:
There should also exist a `Box2 Intersected(Box2 other)` overload to match the rest of the math APIs.

Reply:
Good Point. Will be added.

Reply:
but `Intersected` sounds like it would return `bool` not another `Box`. Name should be `Intersect` then.

Reply:
`Intersects` is the name I would expect on a function that returns a `bool`.